src/fsharp/LexFilter.fs

```diff
@@ -399,39 +399,62 @@ type LexbufState(startPos: Position,
     member x.EndPos = endPos
     member x.PastEOF = pastEOF
 
-[<Struct>]
-type PositionTuple =
-    val X: Position
-    val Y: Position
-    new (x: Position, y: Position) = { X = x; Y = y }
-
 /// Used to save the state related to a token
 [<Class>]
 type TokenTup = 
-    val Token : token
-    val LexbufState : LexbufState
-    val LastTokenPos: PositionTuple
+    // This is mutable for performance reasons.
+    val mutable Token : token
+    val mutable LexbufState : LexbufState
+    val mutable LastTokenPos: Position
     new (token, state, lastTokenPos) = { Token=token; LexbufState=state;LastTokenPos=lastTokenPos }
```

Comment:
It could, but the small benefit we get with how it's currently written is you know exactly the what the object will capture. The downside of primary constructors is that it's easy for the object to keep hold of instances.

I think having it rewritten like this is probably fine considering we are using pooling and it doesn't necessarily matter what size of the object is.

Reply:
I'd rather keep it the way it is actually. I prefer to have the yellow highlighting of the field in IDE tooling letting me know the field is mutable; `TokenTup` is treated as though it's an immutable object, but it's not due to pooling.

Reply:
Worth noting that tooltips and IntelliSense show that it has a setter now if you make it a property, but your rationale is fine